src/site/asciidoc/manual/cloud.adoc

```diff
@@ -0,0 +1,560 @@
+| Kafka Local JSON            |          |           |           |
+| - sendSync true             | 58.46    | 38.55     | 19.59     | 19.01
+| - sendSync false            | 9.8      | 10.8      | 12.23     | 11.36
+| Console                     |          |           |           |
+| - JSON / Kubernetes         | 3.03     | 3.11      | 3.04      | 2.51
+| - JSON                      | 2.80     | 2.74      | 2.54      | 2.35
+| - Docker fluentd driver     | 10.65    | 9.92      | 10.42     | 10.27
+| Rolling File                |          |           |           |
+| - RFC5424                   | 1.65     | 0.94      | 1.22      | 1.55
+| - JSON                      | 1.90     | 0.95      | 1.57      | 1.94
+| TCP - Fluent Bit - JSON     | 2.34     | 2.167     | 1.67      | 2.50
+| Async Logger                |          |           |           |
+| - TCP - Fluent Bit - JSON   | 0.90     | 0.58      | 0.36      | 0.48
+| - Console - JSON            | 0.83     | 0.57      | 0.55      | 0.61
+| - Flume Avro - 1000 - JSON  | 0.76     | 0.37      | 0.45      | 0.68
+|===========================================================================
+
+
+Notes:
+1. Flume Avro - Buffering is controlled by the batch size. Each send is complete when the remote
+acknowledges the batch was written to its channel. These number seem to indicate Flume Avro could
+benefit from using a pool of RPCClients, at least for a batchSize of 1.
+1. Flume Embedded - This is essentially asynchronous as it writes to an in-memory buffer. It is
+unclear why the performance isn't closer to the AsyncLogger results.
+1. Kafka was run in standalone mode on the same laptop as the application. See  sendSync set to true
+requires waiting for an ack from Kafka for each log event.
+1. Console - System.out is redirected to a file by Docker. Testing shows that it would be much
+slower if it was writing to the terminal screen.
+1. Rolling File - Test uses the default buffer size of 8K.
+1. TCP to Fluent Bit - The Socket Appender uses a default buffer size of 8K.
+1. Async Loggers - These all write to a circular buffer and return to the application. The actual
+I/O will take place on a separate thread. If writing the events is performed more slowly than
+events are being created eventually the buffer will fill up and logging will be performed at
+the same pace that log events are written.
+
+== Logging Recommendations
+
+
+1. Use asynchronous logging unless guaranteed delivery is absolutely required. As
+the performance numbers show, so long as the volume of logging is not high enough to fill up the
+circular buffer the overhead of logging will almost be unnoticeable to the application.
+1. If overall performance is a consideration or you require multiline events such as stack traces
+be processed properly then log via TCP to a companion container that acts as a log forwarder or directly
+to a log aggregator as shown above in [Logging with ELK](#ELK). Use the
```

Comment:
Link is broken due to Markdown formatting.

Reply:
fixed

Reply:
There was still something missing. Fixed in 117e004e6e691c0ce943a97d276f0f415083254e.